contributing/topics/schema-design-considerations.md

```diff
@@ -257,3 +257,159 @@ Would be better exposed as the following resources:
+	MaxItems: 20,
+	Elem: &pluginsdk.Schema{
+		Type:         pluginsdk.TypeString,
+		ValidateFunc: validation.StringIsNotEmpty,
+	},
+},
+```
+
+## Required fields in Azure Portal vs API documentation
+
+Fields marked as required in the Azure Portal (indicated by `*`) should be defined as `Required` in Terraform, unless the API accepts the request without them and still functions.
+
+## AtLeastOneOf validation for TypeList fields
+
+When a `pluginsdk.TypeList` block has no required nested fields, `AtLeastOneOf` must be set on the optional fields to ensure at least one is specified.
+
+```go
+"setting": {
+    Type:     pluginsdk.TypeList,
+    Optional: true,
+    MaxItems: 1,
+    Elem: &pluginsdk.Resource{
+        Schema: map[string]*pluginsdk.Schema{
+            "linux": {
+                Type:         pluginsdk.TypeList,
+                Optional:     true,
+                Elem:         osSchema(),
+                AtLeastOneOf: []string{"setting.0.linux", "setting.0.windows"},
+            },
+            "windows": {
+                Type:         pluginsdk.TypeList,
+                Optional:     true,
+                Elem:         osSchema(),
+                AtLeastOneOf: []string{"setting.0.linux", "setting.0.windows"},
+            },
+        },
+    },
+}
+```
+
+## Validation
+
+String arguments must be validated. Use `StringNotEmpty` at a minimum. Validate `name` fields for length and allowed characters. Use `commonids` or SDK-specific functions for Resource IDs. Ensure formats like dates, IPs, and URIs are validated.
+
+Numeric arguments should specify a valid range.
+
+```go
+"description": {
+	Type:         pluginsdk.TypeString,
+	Optional:     true,
+	ValidateFunc: validation.StringIsNotEmpty,
+},
+
+"name": {
+	Type:     pluginsdk.TypeString,
+	Required: true,
+	ForceNew: true,
+	ValidateFunc: validation.StringMatch(
+		regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9.\-_]{0,79}$`),
+		"`name` must be between 1 and 80 characters. It must start with an alphanumeric character and can contain alphanumeric characters, dots (.), hyphens (-), and underscores (_).",
+	),
+},
+
+"subnet_id": {
+	Type:         pluginsdk.TypeString,
+	Required:     true,
+	ValidateFunc: commonids.ValidateSubnetID,
+},
+
+"sim_policy_id": {
+	Type:         pluginsdk.TypeString,
+	Optional:     true,
+	ValidateFunc: simpolicy.ValidateSimPolicyID,
+},
+
+"extensions_time_budget": {
+	Type:         pluginsdk.TypeString,
+	Optional:     true,
+	Default:      "PT1H30M",
+	ValidateFunc: validate.ISO8601DurationBetween("PT15M", "PT2H"),
+},
+
+"output_blob_uri": {
+	Type:         pluginsdk.TypeString,
+	Optional:     true,
+	ValidateFunc: validation.IsURLWithHTTPS,
+},
+
+"ip_address": {
+	Type:         pluginsdk.TypeString,
+	Required:     true,
+	ValidateFunc: azValidate.IPv4Address,
+},
+
+"storage_size_in_gb": {
+	Type:         pluginsdk.TypeInt,
+	Optional:     true,
+	ValidateFunc: validation.IntBetween(32, 16384),
+},
+
+"percentage_filter_value": {
```

Comment:
this contradicts the above guidance for `sampling_percentage` and should be
```suggestion
"filter_value_percentage": {
```
?